### DISCUSSION TITLE
nRF52 SoC flash usage in smp_svr example

### DISCUSSION THREAD

**Discussion by Zheckiss** [zephyrproject-rtos/zephyr#60659] [Q&A]
smp_svr example mounts littleFS file system in SoC flash for OTA DFU function. Can I use that file system to save my variables? When I use fs_open() function there is an error: fs: invalid file name!! Sorry, I don't know, how to make readable code here...
`#define STORAGE_PARTITION_LABEL	storage_partition
#define STORAGE_PARTITION_ID	FIXED_PARTITION_ID(STORAGE_PARTITION_LABEL)

/* Define an example stats group; approximates seconds since boot. */
STATS_SECT_START(smp_svr_stats)
STATS_SECT_ENTRY(ticks)
STATS_SECT_END;

/* Assign a name to the `ticks` stat. */
STATS_NAME_START(smp_svr_stats)
STATS_NAME(smp_svr_stats, ticks)
STATS_NAME_END(smp_svr_stats);

/* Define an instance of the stats group. */
STATS_SECT_DECL(smp_svr_stats) smp_svr_stats;

#ifdef CONFIG_MCUMGR_GRP_FS
FS_LITTLEFS_DECLARE_DEFAULT_CONFIG(cstorage);
static struct fs_mount_t littlefs_mnt = {
	.type = FS_LITTLEFS,
	.fs_data = &cstorage,
	.storage_dev = (void *)STORAGE_PARTITION_ID,
	.mnt_point = "/lfs1"
};
#endif

int main(void)
{
	
	struct fs_file_t file;
	struct fs_dir_t dir;
	struct fs_statvfs sbuf;
	char f_Name[] = "/vars.dat";
	uint8_t test_Bytes[] = {0x01,0x02,0x03,0x04};
	

	NRF_POWER->DCDCEN = false;
	NRF_UICR->NFCPINS = 0;  // NFC pins as GPIOs
	
	LOG_LEVEL_SET(LOG_LEVEL_INF);
	int rc = STATS_INIT_AND_REG(smp_svr_stats, STATS_SIZE_32,
				    "smp_svr_stats");

	if (rc < 0) {
		LOG_ERR("Error initializing stats system [%d]", rc);
	}
    
	/* Register the built-in mcumgr command handlers. */

	rc = fs_mount(&littlefs_mnt);
	if (rc < 0) {
		LOG_ERR("Error mounting littlefs [%d]", rc);
	} else {
		LOG_INF("result [%d]", rc);
	}
	k_sleep(K_MSEC(50));

	
//fs_dir_t_init(&dir);
fs_file_t_init(&file);
rc = fs_statvfs(&littlefs_mnt.mnt_point, &sbuf);
	if (rc < 0) {
		printk("FAIL: statvfs: %d\n", rc);
		return;
	}

	printk("%s: bsize = %lu ; frsize = %lu ;"
	       " blocks = %lu ; bfree = %lu\n",
	       littlefs_mnt.mnt_point,
	       sbuf.f_bsize, sbuf.f_frsize,
	       sbuf.f_blocks, sbuf.f_bfree);

rc = fs_open(&file, f_Name, FS_O_CREATE|FS_O_RDWR);
	if (rc < 0) {
		LOG_ERR("Error of opening file [%d]", rc);
	} else {
		LOG_INF("File opened ok!");
	}
rc = fs_seek(&file, 0, FS_SEEK_SET);
	if (rc < 0) {
		LOG_ERR("FAIL: seek %s: %d", f_Name, rc);
	}
ssize_t rf = fs_write(&file, test_Bytes, 4);
	if (rf < 0) {
		LOG_ERR("Error of writing file [%d]", rf);
	} else {
		LOG_INF("File written ok! [%d]", rf);
	}
rc = fs_close(&file);
	if (rc < 0) {
		LOG_ERR("Error of closing file [%d]", rc);
	}

#ifdef CONFIG_MCUMGR_TRANSPORT_BT
	start_smp_bluetooth_adverts();
#endif

	while (1) {
		k_sleep(K_MSEC(1000));
		STATS_INC(smp_svr_stats, ticks);
	}
	return 0;
}
`

**Comment by thedjnK** [comment]
You need to reformat the code between a \`\`\` at the top and \`\`\` at the bottom. Also, don't do things like touch the UICR, this is not correct, there are Kconfig options to set these values
